### DISCUSSION TITLE
Is it possible to get the metadata of all audio files on the local storage/sd card

### DISCUSSION THREAD

**Discussion by ismailbazookka** [expo/expo#18409] [Archived: Questions]
I'm trying to build a music player with expo but I couldn't find how to retrieve the audio files and read them anywhere.

**Comment by barthap** [comment]
On iOS, there's no possibility for listing audio items yet (I'm going to implement it some day). On Android, it is possible with [`expo-media-library`](https://docs.expo.dev/versions/v46.0.0/sdk/media-library/):
```js
import * as MediaLibrary from `expo-media-library`;

const { assets } = await MediaLibrary.getAssetsAsync({ mediaType: 'audio', first: 20 });
```

---

If you're going to create a single-song player, where one or more songs are picked by system UI, you can check my experimental app: [expo-mega-demo](https://github.com/barthap/expo-mega-demo) which implements this feature (for a single song, but multiple can be selected with 3rd party [expo-music-picker](https://github.com/barthap/expo-music-picker)).
